### PR TITLE
Update maintainers guides

### PR DESCRIPTION
**Pull request by seisman**
**Preview**:  https://pygmt-dev--2916.org.readthedocs.build/en/2916/maintenance.html

### CODE REVIEW

File: doc/maintenance.md [hunk 1]
```diff
@@ -86,33 +87,34 @@ The main advantages of this are:
...
+- `ci_docs.yml`: Build documentation on Linux/macOS/Windows and deploy to GitHub
+- `ci_doctest.yaml`: Run all doctests on Linux/macOS/Windows
+- `cache_data.yaml`: Cache GMT remote data files and uplodas as artifacts
+- `publish-to-pypi.yml`: Publish archives to PyPI and TestPyPI
+- `release-drafter.yml`: Draft the next release notes
+- `release-baseline-images.yml`: Upload the ZIP archive of baseline images as a release asset
+- `format-command.yml`: Format the codes using slash command
+- `dvc-diff.yml`: Report changes in test images
+- `slash-command-dispatch.yml`: Support slash commands in pull requests
+- `benchmarks.yml`: Benchmarks the execution speed of tests to track performance of PyGMT functions
```
Comment (weiji14) [inline doc/maintenance.md:109]: Sort this list alphabetically perhaps?

Comment (seisman) [reply doc/maintenance.md]: Done in fc4f9ec1.

**Review by weiji14 — APPROVED** [review]

**Comment by seisman** [comment]
@weiji14 Do you want to add a section like "Continuous Benchmark" to this guide?

**Comment by weiji14** [comment]
> @weiji14 Do you want to add a section like "Continuous Benchmark" to this guide?

I did think about it when writing the docs in #2908, but wasn't sure what to include yet in terms of guidelines. Maybe after a few weeks/months when we've got some experience with benchmarking, then we can add some words in.

**Comment by seisman** [comment]
> > @weiji14 Do you want to add a section like "Continuous Benchmark" to this guide?
> 
> I did think about it when writing the docs in #2908, but wasn't sure what to include yet in terms of guidelines. Maybe after a few weeks/months when we've got some experience with benchmarking, then we can add some words in.

Sounds good.

File: doc/maintenance.md [hunk 2]
```diff
@@ -263,8 +267,9 @@ If changes need to be done manually, you can:
 1. Fork the [pygmt feedstock repository](https://github.com/conda-forge/pygmt-feedstock) if
    you haven't already. If you have a fork, update it.
 2. Update the version number and sha256 hash on `recipe/meta.yaml`. You can get the hash
-   from the PyPI "Download files" section.
+   from the [PyPI "Download files" section](https://pypi.org/project/pygmt/#files).
 3. Add or remove any new dependencies (most are probably only `run` dependencies).
-4. Make a new branch, commit, and push the changes **to your personal fork**.
-5. Create a PR against the original feedstock main.
-6. Once the CI tests pass, merge the PR or ask a maintainer to do so.
+4. Make sure the mimimum support versions of all dependencies are correctly pinned.
```
Comment (yvonnefroehlich) [inline doc/maintenance.md:272]:
```suggestion
4. Make sure the minimum support versions of all dependencies are correctly pinned.
```

**Review by yvonnefroehlich — APPROVED** [review]

**Pull request merged**